### DISCUSSION TITLE
Cloudlog and sync LoTW

### DISCUSSION THREAD

**Discussion by myhobby2** [magicbug/Cloudlog#2944] [Q&A]
Until early this year sync with LoTW was working but today I can't get it sync-ed anymore. Deleted the .p12 certificate and reinstalled it and now states: status = valid / last upload = not synced. When I use the button MANUAL SYNC nothing improves. 

Just updated to the latest release today. Does anyone has encountered a similar situation (and is there a solution?)
![Lotw](https://github.com/magicbug/Cloudlog/assets/49870972/07f4758f-31bf-4e8d-b948-862be3337913)

**Comment by magicbug** [comment]
The fact the callsign and dxcc is empty suggests a problem. I'd suggest deleting it from the system, making sure tQSL is up to date on your computer and then exporting a new p12

**Reply by myhobby2** [reply]
That is done by my with a purpose. There is a callsign and DXCC available ;-) and I already  did beforehand what you suggested.

**Reply by magicbug** [reply]
Is the callsign and DXCC displayed now within the table now?

If not, we will need to enable logging /application/config/config.php and then reupload the certificate and provide the output in /application/logs so that it can be investigated further.

**Comment by geogarden** [comment]
I have the same problem, it doesn't show me the status synchronized

![Zrzut ekranu z 2024-01-30 10-32-33](https://github.com/magicbug/Cloudlog/assets/33130991/b1429c43-7042-4955-b4be-3e1d9b939752)

**Reply by magicbug** [reply]
This is a separate problem have you tried to manually upload to LoTW via the button that says manual sync?

**Reply by myhobby2** [reply]
That is exactly my problem ;-)

**Reply by magicbug** [reply]
@myhobby2 this isn't the problem you first reported. as can be seen by both screenshots you where missing callsign and dxcc information without that it shows that Cloudlog couldn't process the certificate.

**Reply by magicbug** [reply]
Please do the following

 enable logging /application/config/config.php and then reupload the certificate and provide the output in /application/logs so that it can be investigated further. also run manual sync and provide the logs without this its impossible to help further.

**Reply by myhobby2** [reply]
ERROR - 2024-01-30 14:43:17 --> Severity: Warning --> fopen(./uploads/lotw/on3???-2024-01-30-14-43-17-cloudlog.tq8): Failed to open stream: Permission denied /var/www/cloudlog/application/controllers/Lotw.php 275
ERROR - 2024-01-30 14:43:17 --> Severity: error --> Exception: fwrite(): Argument #1 ($stream) must be of type resource, bool given /var/www/cloudlog/application/controllers/Lotw.php 276

I editted my callsign because it is not relevant to the issue.

**Reply by myhobby2** [reply]
As extra info: Linux cloudlog 6.1.0-17-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.69-1 (2023-12-30) x86_64

**Reply by magicbug** [reply]
Looks like /uploads folder permissions are incorrectly setup

**Reply by myhobby2** [reply]
your were right but let me explain what I did
/var/www/cloudlog/uploads/ was 0775 and I changed it to 0777. Only for this directory

I never touched any of these permissions, knowing what the consequences are. I did do twice an upgrade via the script in /var/www/cloudlog/update_cloudlog.sh. No idea, but just a clue. 

It works again after this permission change, but you can help me by providing the correct permission ;-)

**Reply by myhobby2** [reply]
Thanks for the help before I forget to mention this! Very much appreciated
